recordval.go

```diff
@@ -31,39 +29,30 @@ func (r *Value) Walk(rv Visitor) error {
 	return Walk(r.Type, r.Bytes, rv)
 }
 
-// Slice returns the encoded zcode.Bytes corresponding to the indicated
-// column or an error if a problem was encountered.
-func (r *Value) Slice(column int) (zcode.Bytes, error) {
+func (r *Value) nth(column int) zcode.Bytes {
 	var zv zcode.Bytes
 	for i, it := 0, r.Bytes.Iter(); i <= column; i++ {
 		if it.Done() {
-			return nil, ErrMissing
+			return nil
 		}
 		zv = it.Next()
 	}
-	return zv, nil
+	return zv
 }
 
 func (r *Value) Columns() []Column {
 	return TypeRecordOf(r.Type).Columns
 }
 
-// Value returns the indicated column as a Value.  If the column doesn't
-// exist or another error occurs, the nil Value is returned.
-func (r *Value) ValueByColumn(col int) Value {
-	zv, err := r.Slice(col)
-	if err != nil {
-		return Value{}
-	}
-	return Value{r.Columns()[col].Type, zv}
-}
-
-func (r *Value) ValueByField(field string) (Value, error) {
-	col, ok := r.ColumnOfField(field)
-	if !ok {
-		return Value{}, ErrMissing
+func (v *Value) DerefByColumn(col int) *Value {
+	if v != nil {
+		if bytes := v.nth(col); bytes != nil {
+			v = &Value{v.Columns()[col].Type, bytes}
```

Comment:
This would panic if Columns is nil (because the value is a record) or not the right size, I think we want to turns nil here right?

Reply:
Yes, will fix.

Reply:
Fixed in ColumnOfField, which fixes it here.

Reply:
If Value is an array or set you'll get a panic here.

Reply:
Or should this private and only called by record-y functions?

Reply:
Table/text writer call DerefByColumn but only with records.  My point is for this method, it needs to be a record as implied by the name.  We will document all this and improve the names further in a subsequent PR.